sql/parser/expr.go

```diff
@@ -241,20 +241,31 @@ func (NullVal) String() string {
 	return fmt.Sprintf("NULL")
 }
 
-// QualifiedName is a dot separated list of names.
-type QualifiedName []string
+// QualifiedName is a base name and an optional indirection expression.
+type QualifiedName struct {
+	Base     Name
+	Indirect Indirection
+}
 
 // Database returns the database portion of the name.
-func (n QualifiedName) Database() string {
-	if len(n) > 1 {
-		return n[0]
+func (n *QualifiedName) Database() string {
+	if len(n.Indirect) > 0 {
+		return n.Base.String()
 	}
 	return ""
 }
 
 // Table returns the table portion of the name.
-func (n QualifiedName) Table() string {
-	return n[len(n)-1]
+func (n *QualifiedName) Table() string {
```

Comment:
this function's correctness still depends on QualifiedName not referring to a column, is that right? maybe it deserves a comment.